### PR TITLE
feat: Added basic Anchor component

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- [x] Related to Abstract designs: https://app.abstract.com/projects/a4a24600-c8c9-11e8-a177-81a5fa367641/branches/328ef9fe-09cf-43f4-8dd1-40947f5d4fb4/commits/3a51a13f42daf13efbfd15dc96f34ad00a72052d/files/FA9AF2C4-35B2-4183-8E93-A9D84A525613/layers/7598FF01-E0C7-4FAD-A505-B6136FA4DC91?mode=design
- [x] Related to JIRA ticket: A11Y-415
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

We want `text-decoration: underline` on basic inline links as an opt-in behavior when they're not created by markdown. This moves that part of the existing markdown component override into a top-level link.

Screenshot of plain text with a purple underlined link

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.48-alpha.bc07e7.0<br/>@codecademy/gamut@11.0.1-alpha.bc07e7.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.48-alpha.93a24b.0<br/>@codecademy/gamut@11.0.1-alpha.93a24b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f18b8c4bb3cc32cb4f2c423--codecademy-styleguide.netlify.app>https://5f18b8c4bb3cc32cb4f2c423--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f18b8c4bb3cc32cb4f2c423>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f18b97ccf096bed6b39ba01--codecademy-styleguide.netlify.app>https://5f18b97ccf096bed6b39ba01--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f18b97ccf096bed6b39ba01>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.51-alpha.ce8639.0<br/>@codecademy/gamut@12.0.2-alpha.ce8639.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f18bd5253314499f21ff1a9--codecademy-styleguide.netlify.app>https://5f18bd5253314499f21ff1a9--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f18bd5253314499f21ff1a9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.51-alpha.57fa58.0<br/>@codecademy/gamut@12.0.2-alpha.57fa58.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f18c29dbb3cc35e90f2c692--codecademy-styleguide.netlify.app>https://5f18c29dbb3cc35e90f2c692--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f18c29dbb3cc35e90f2c692>Deploy Logs</a></p>

File: packages/gamut/src/Anchor/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  as?: ChildComponentDescriptor;
+
+  /**
+   * Props to pass directly to the container element.
+   *
+   * @remarks We would love to properly type this with generics, but cannot yet.
+   * @see https://github.com/Codecademy/client-modules/pull/270#discussion_r270917147
+   * @see https://github.com/Microsoft/TypeScript/issues/21048
+   */
+  asProps?: any;
```
Comment (codecaaron) [inline packages/gamut/src/Anchor/index.tsx:20]: I am assuming that this to enable usage with like `redux-first-router-link` without having to type it for both right?

Not that CSS in JS won't be a massively breaking change but I'm wondering for these single level elements if we should try to be explicit with expected APIs where possible to limit refactoring RE: 
https://emotion.sh/docs/styled#customizing-prop-forwarding

Comment (codecaaron) [reply packages/gamut/src/Anchor/index.tsx:20]: Also the two probable uses of this component 'Link' and 'a' I think are an interesting discriminating union case.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.1-alpha.4a64cb.0<br/>@codecademy/gamut@12.2.1-alpha.4a64cb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2009d3d6b22875849915fa--gamut-preview.netlify.app>https://5f2009d3d6b22875849915fa--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2009d3d6b22875849915fa>Deploy Logs</a></p>

File: packages/gamut/src/Anchor/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Anchor/index.tsx:20]: Heh, I'm getting some of the discriminated unions excitement as a runoff...

That would be a good case, but I'm thinking we don't want to take a redux dependency here (literally or in a types only theoretical way) to set up explicit props. Ideally we'll be able to use this for arbitrarily any child, including our custom `TrackLink`s during conversion.

Also, once nice plug for `asProps`, if we're gonna break the heck out of everything anyway, it's pretty straightforward to go from it to a base prop spread. 

So... Thoughts? What so you prefer we do?

Comment (codecaaron) [reply packages/gamut/src/Anchor/index.tsx:20]: Reasonable to me :D

**Pull request merged**